lib/ansible/cli/__init__.py

```diff
@@ -275,7 +275,7 @@ def init_parser(self):
                 ansible.arguments.option_helpers.add_runas_options(self.parser)
                 self.parser.add_option('--my-option', dest='my_option', action='store')
         """
-        self.parser = opt_help.create_base_parser(usage=usage, desc=desc, epilog=epilog)
+        self.parser = opt_help.create_base_parser(usage=usage, desc=desc, epilog=epilog, prog=os.path.basename(self.args[0]))
```

Comment:
@abadger I'm not sure if this is a good default. If `args` is an empty list (as it was in the unit test), this raises an `IndexError`. Should I do some sanity checking or `args` or will that never be the case during normal operation?

Reply:
During normal operation, it would never be empty (A program is always invoked with at least its filename as an argument.)

Perhaps we should sanity check in the ```CLI``` constructor.

``` python
def __init__(self, args, callback=None):
    if not args:
        raise ValueError('Require a non-empty args list')
```

OR

``` python
def __init__(self, args, callback=None):
    if not args:
        args = ['ansible test program']
```

Reply:
Sounds good. I'll sanity check in the `CLI` constructor.